lib/ansible/galaxy/api.py

```diff
@@ -817,8 +817,20 @@ def get_collection_version_metadata(self, namespace, name, version):
 
         signatures = data.get('signatures') or []
 
+        download_url_info = urlparse(data['download_url'])
+        if not download_url_info.scheme and not download_url_info.path.startswith('/'):
+            # galaxy does a lot of redirects, with much more complex pathing than we use
+            # within this codebase, without updating _call_galaxy to be able to return
+            # the final URL, we can't reliably build a relative URL.
+            raise AnsibleError(f'Invalid non absolute download_url: {data["download_url"]}')
+
+        if download_url_info.scheme:
+            download_url = data['download_url']
+        else:
+            download_url = urljoin(self.api_server, data['download_url'])
```

Comment:
I think this equivalent since we can expect URLs with scheme or abs path?

```suggestion
        download_url = urljoin(self.api_server, data['download_url'])
```

Reply:
Yeah, I think I was doing a pre-optimization, although I doubt it's necessary.